packages/server/src/service/signup.ts

```diff
@@ -0,0 +1,84 @@
+import * as admin from 'firebase-admin'
+import { StateInfo, ContactMethod } from '../common'
+import { Letter } from './letter'
+import { toPdfBuffer } from './pdf'
+import { sendSignupEmail } from './mg'
+import { FirestoreService } from './firestore'
+import { storageFileFromId, StorageFile } from './storage'
+import { TwilioResponse, RichStateInfo } from './types'
+import { sendFaxes } from './twilio'
+const firestoreService = new FirestoreService()
+
+
+export const sendAndStoreSignup = async (
+  info: StateInfo,
+  method: ContactMethod,
+  id: string,
+  resendReasonAndOriginalDate?: string,
+) => {
+  const letter = new Letter(info, method, id, resendReasonAndOriginalDate)
+  const pdfBuffer = await toPdfBuffer(letter.render('html'), await letter.form)
+  const isResend = resendReasonAndOriginalDate !== undefined
+  if (isResend && info?.id === undefined) {
+    throw new Error('When resending emails, StateInfo should be RichStateInfo with id')
+  }
+
+  // Send email (perhaps only to voter)
+  const mgResponse = await sendSignupEmail(
+    letter,
+    info.email,
+    method.emails,
+    { pdfBuffer, id },
+  )
+  if (!mgResponse || mgResponse.message !== 'Queued. Thank you.') {
+    console.error('Error sending email for ' + info.id + '. Exiting.')
+    console.error(mgResponse)
+    return
+  }
+
+  // Upload PDF
+  let file: StorageFile
+  if (isResend) {
+    // When resending signups Needs to get how many times this file was resent
+    const { id, resends } = info as RichStateInfo
+
+    const numResends = resends ? resends.length : 0
+    file = storageFileFromId(`${id}-${numResends}`)
+    await file.upload(pdfBuffer)
+  } else {
+    file = storageFileFromId(id)
+    await file.upload(pdfBuffer)
+  }
+
+  // Send faxes
+  let twilioResponses: TwilioResponse[] = []
+  if (method.faxes.length > 0) {
+    const uri = await file.getSignedUrl(24 * 60 * 60 * 1000)
+    const resposnes = await sendFaxes(uri, method.faxes)
+    twilioResponses = resposnes.map(({url, sid, status}) => ({url, sid, status}))
+
+    for (const twilioResponse of twilioResponses) {
+      if (twilioResponse.status !== "queued") {
+        console.error(`Error sending faxes for ${info.id ?? id}. Exiting.`)
+        console.error(twilioResponse)
+        return
+      }
+    }
+  }
+
+  if (isResend) {
+    // Update voter data in firestore.
+    const resend = { dateTime: new Date().getTime(), reason: resendReasonAndOriginalDate }
+    await firestoreService
+      .db
+      .collection('StateInfo')
+      // We'd throw if info.id is undefined at the beginning of this function
```

Comment:
To fix this, you should use `id` not `info.id`.  It doesn't store type inferences on class properties (presumably because they are mutable).